### PR TITLE
Fix capture picking bad robots

### LINKED ISSUE #375: Make capture choose a robot close to the intercept point
Capture moves the robot to an intercept point along the ball's trajectory if it is moving, so the role_requirements() method should specify that point as the preferred bot location, not the ball's current location.  This may be best as a custom cost function since it's robot-dependent.

### PR DESCRIPTION
**Pull request by jcarn**
Closes #375

### CODE REVIEW

**Comment by jcarn** [comment]
The role req stuff is @mmurley3 PR that hasn't merged yet, but it was useful

File: soccer/gameplay/skills/capture.py [hunk 1]
```diff
@@ -152,5 +132,37 @@ def role_requirements(self):
...
         # try to be near the ball
         if main.ball().valid:
-            reqs.destination_shape = main.ball().pos
+            reqs.cost_func = lambda r: reqs.position_cost_multiplier * find_robot_intercept_ball(
+                r).dist_to(r.pos)
+
         return reqs
+
+
+def find_robot_intercept_ball(robot):
```
[the remark this answers is not in the document]
Comment (jcarn) [reply soccer/gameplay/skills/capture.py]: It specifically uses the ball, but in the future having a function that could find an intercept with both a robot and a ball (and anything else that moves?) could be useful so I renamed it

[the remark this answers is not in the document]
Comment (JNeiger) [reply soccer/gameplay/skills/capture.py]: That's a good point. I think I was comparing it to the previous version of that function.

File: soccer/gameplay/skills/capture.py [hunk 2]
```diff
@@ -73,32 +76,9 @@ def bot_in_front_of_ball(self):
...
-            ball_time = evaluation.ball.rev_predict(main.ball().vel, dist)
-            robotDist = (pos - self.robot.pos).mag() * 0.6
-            bot_time = robocup.get_trapezoidal_time(robotDist, robotDist, 2.2,
-                                                    1, self.robot.vel.mag(), 0)
-
-            if bot_time < ball_time:
-                break
-
-        return pos
+        return find_robot_intercept_ball(self.robot)
```
Comment (JNeiger) [inline soccer/gameplay/skills/capture.py]: Don't forget this one either

File: soccer/gameplay/skills/capture.py [hunk 3]
```diff
@@ -152,5 +132,37 @@ def role_requirements(self):
...
+
+
+def find_robot_intercept_point(robot):
+    if robot is None:
+        return main.ball().pos
+    approach_vec = approach_vector(robot)
+    # sample every 5 cm in the -approach_vector direction from the ball
+    pos = None
+    for i in range(50):
+        dist = i * 0.05
```
Comment (jgkamat) [inline soccer/gameplay/skills/capture.py]: I /really/ don't like magic numbers, can you make some global constants for these

File: soccer/gameplay/skills/capture.py [hunk 4]
```diff
@@ -152,5 +132,37 @@ def role_requirements(self):
...
         # try to be near the ball
         if main.ball().valid:
-            reqs.destination_shape = main.ball().pos
+            reqs.cost_func = lambda r: reqs.position_cost_multiplier * find_robot_intercept_point(
+                r).dist_to(r.pos)
+
         return reqs
+
+
+def find_robot_intercept_point(robot):
```
Comment (jgkamat) [inline soccer/gameplay/skills/capture.py]: This function is giving me a ton of deja vu, does it exist somewhere else already? I could just be misthinking though.

File: soccer/gameplay/skills/capture.py [hunk 5]
```diff
@@ -152,5 +132,37 @@ def role_requirements(self):
...
+    approach_vec = approach_vector(robot)
+    # sample every 5 cm in the -approach_vector direction from the ball
+    pos = None
+    for i in range(50):
+        dist = i * 0.05
+        pos = main.ball().pos + approach_vec * dist
+        # how long will it take the ball to get there
+        ball_time = evaluation.ball.rev_predict(main.ball().vel, dist)
+        robotDist = (pos - robot.pos).mag() * 0.6
+        bot_time = robocup.get_trapezoidal_time(robotDist, robotDist, 2.2, 1,
```
Comment (jgkamat) [inline soccer/gameplay/skills/capture.py]: The 2.2 constant seems pretty important (max speed?). We should pull this from motion constratints at the very least

**Pull request merged**